Session/Conversations/ConversationVC.swift

```diff
@@ -2001,9 +1996,64 @@ final class ConversationVC: BaseVC, SessionUtilRespondingViewController, Convers
         }
     }
     
+    
+    func getNewestMessageViewModel() -> MessageViewModel? {
+        // Note: For the 'tableVisualBottom' we remove the 'Values.mediumSpacing' as that is the distance
+        // the table content appears above the input view
+        let tableVisualBottom: CGFloat = (tableView.frame.maxY - (tableView.contentInset.bottom - Values.mediumSpacing))
+
+        let visibleIndexPaths: [IndexPath]? = self.tableView.indexPathsForVisibleRows
+        let messagesSection: Int? = visibleIndexPaths?.first(where: { self.viewModel.interactionData[$0.section].model == .messages })?.section
+        let newestCellViewModel: MessageViewModel? = visibleIndexPaths?
+            .sorted()
+            .filter { $0.section == messagesSection }
+            .compactMap { indexPath -> (frame: CGRect, cellViewModel: MessageViewModel)? in
+                guard let cell: VisibleMessageCell = tableView.cellForRow(at: indexPath) as? VisibleMessageCell else {
+                    return nil
+                }
+
+                return (
+                    view.convert(cell.frame, from: tableView),
+                    self.viewModel.interactionData[indexPath.section].elements[indexPath.row]
+                )
+            }
+            // Exclude messages that are partially off the bottom of the screen
+            .filter { $0.frame.maxY <= tableVisualBottom }
+            .last?
+            .cellViewModel
+
+        return newestCellViewModel
+    }
+    
     // MARK: - SessionUtilRespondingViewController
     
     func isConversation(in threadIds: [String]) -> Bool {
         return threadIds.contains(self.viewModel.threadData.threadId)
     }
 }
+
+struct ConversationReplyNavigationStack {
```

Comment:
I couldn't find a better place to add this, but let me know if there is one, and I'll move it :)